src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs

```diff
@@ -300,6 +303,14 @@ public async ValueTask<SyncedStateLease<ReadPosition>> LeaseReadPositionState(Ch
         return result;
     }
 
+    public async ValueTask<ComputedStateLease<Range<long>>> LeaseChatIdRangeState(ChatId chatId, CancellationToken cancellationToken)
+    {
+        var lease = await _chatIdRangeStates.Rent(chatId, cancellationToken).ConfigureAwait(false);
+        var result = new ComputedStateLease<Range<long>>(lease);
+        await result.WhenSynchronized(cancellationToken).ConfigureAwait(false);
```

Comment:
There is no timeout here.

Reply:
I don't get you, we should wait for the GetIdRange call, what do you want to return if we won't wait for the call completion?